Packet++/header/RadiusLayer.h

```diff
@@ -6,6 +6,14 @@
 
 /// @file
 
+#ifndef PCPP_NOEXCEPT
```

Comment:
is there a particular reason to use `noexcept` here? Usually I try not to use C++11 features unless absolutely necessary

Reply:
The compiler can use this information to optimize calls to the function. In old C++ the throw() is used for this purpose. For inlined functions noexcept is unnecessary I guess.

Reply:
I think it's better to make `isDataValid` inline. But following code should be moved into the header file:
```
#if defined(WIN32) || defined(WINx64) //for using ntohl, ntohs, etc.
#include <winsock2.h>
#elif LINUX
#include <in.h> //for using ntohl, ntohs, etc.
#elif MAC_OS_X
#include <arpa/inet.h> //for using ntohl, ntohs, etc.
#endif

```
 What You think?

Reply:
I prefer not to move this code to the header file, it creates a lot of issues when linking app to PcapPlusPlus library.

Regarding `noexcept` - I think we should either keep only `throw()`, or remove `PCPP_NOEXCEPT` altogether. I don't see a point here in keeping both `throw()` (for older C++ versions) and `noexcept`.

BTW, I just noticed that there is no NULL check in `isDataValid()`, meaning if `udpData` is NULL, you might end up with seg fault. Am I missing anything?

Reply:
The function isDataValid is corrected.

throw()/noexcept specification is excluded from sources because throw() is removed in C++20.